sdk/src/androidTest/java/com/mongodb/stitch/StitchClientTest.kt

```diff
@@ -0,0 +1,218 @@
+                        "resetPasswordUrl" to FakeDeferUrl
+                ),
+                "oauth2/google" to mapOf(
+                        "clientId" to FakeGoogleClientId,
+                        "metadataFields" to arrayOf(
+                                FakeMetadataFieldEmail,
+                                FakeMetadataFieldBirthday
+                        )
+                ),
+                "oauth2/facebook" to mapOf(
+                        "clientId" to FakeFacebookClientId,
+                        "metadataFields" to arrayOf(
+                                FakeMetadataFieldEmail,
+                                FakeMetadataFieldBirthday
+                        )
+                )
+        )).toString()
+
+        /** Mock data for auth object from [StitchClient.logInWithProvider] */
+        val mockAuthData = JSONObject(mapOf(
+                "accessToken" to FakeAccessToken,
+                "refreshToken" to FakeRefreshToken,
+                "userId" to FakeUserId,
+                "deviceId" to FakeDeviceId
+        )).toString()
+
+        /** Mock data for pipeline execution from [StitchClient.executePipeline] */
+        val mockPipelineData = JSONObject(mapOf(
+                "result" to arrayOf(FakePipelineLiteralFoo, FakePipelineLiteralBar)
+        )).toString()
+
+        /** Mock data for user profile from [StitchClient.getUserProfile] */
+        val mockUserData = JSONObject(mapOf(
+                "domainId" to FakeDomainId,
+                "userId" to FakeUserId,
+                "identities" to arrayOf(mapOf(
+                        "id" to FakeUserId,
+                        "provider" to FakeAnonIdentity)
+                ),
+                "data" to null,
+                "type" to FakeUserType
+        )).toString()
+    }
+
+    /** Base context from test runner */
+    private val instrumentationCtx: Context = InstrumentationRegistry.getContext()
+    /** [StitchClient] for this test */
+    private var stitchClient: StitchClient? = null
+
+    @Before
+    fun setup() {
+        // start the mock server
+        RESTMockServerStarter.startSync(AndroidAssetsFileParser(instrumentationCtx), AndroidLogger())
+
+        // instantiate a new StitchClient using a dummy name and the mock baseUrl
+        val stitchClient = StitchClient(instrumentationCtx, "dummy-app", RESTMockServer.getUrl())
+        this.stitchClient = stitchClient
+
+        // clear all instances of the internal [SharedPreferences] to start with a clean slate
+        stitchClient.properties.clear()
+        stitchClient.javaClass.kotlin.declaredMemberProperties.first {
+            it.name == "_preferences"
+        }.also { it.isAccessible = true }.get(stitchClient).also {
+            (it as SharedPreferences).edit().clear().commit()
+        }
+
+        // mock out all calls related to auth
+        RESTMockServer.whenPOST(pathContains("auth")).thenReturnString(mockAuthData)
```

Comment:
I'd like these to be more specific as spec-like such that we know exactly what type of path we are looking for. It may be easy to mock the wrong thing this way.